client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java

```diff
@@ -0,0 +1,214 @@
+                    "\"bucket_span\": 3600,\"is_interim\": " + isInterim + ", \"record_score\": " + recordScore + ", \"probability\": "
+                    + p + "}", XContentType.JSON);
+            bulkRequest.add(indexRequest);
+        }
+    }
+
+    @After
+    public void deleteJob() throws IOException {
+        new MlRestTestStateCleaner(logger, client()).clearMlMetadata();
+    }
+
+    public void testGetBuckets() throws IOException {
+        MachineLearningClient machineLearningClient = highLevelClient().machineLearning();
+
+        {
+            GetBucketsRequest request = new GetBucketsRequest(JOB_ID);
+
+            GetBucketsResponse response = execute(request, machineLearningClient::getBuckets, machineLearningClient::getBucketsAsync);
+
+            assertThat(response.count(), equalTo(241L));
+            assertThat(response.buckets().size(), equalTo(100));
+            assertThat(response.buckets().get(0).getTimestamp().getTime(), equalTo(1533081600000L));
+        }
+        {
+            GetBucketsRequest request = new GetBucketsRequest(JOB_ID);
+            request.setTimestamp("1533081600000");
+
+            GetBucketsResponse response = execute(request, machineLearningClient::getBuckets, machineLearningClient::getBucketsAsync);
+
+            assertThat(response.count(), equalTo(1L));
+            assertThat(response.buckets().size(), equalTo(1));
+            assertThat(response.buckets().get(0).getTimestamp().getTime(), equalTo(1533081600000L));
+        }
+        {
+            GetBucketsRequest request = new GetBucketsRequest(JOB_ID);
+            request.setAnomalyScore(75.0);
+
+            GetBucketsResponse response = execute(request, machineLearningClient::getBuckets, machineLearningClient::getBucketsAsync);
+
+            assertThat(response.count(), equalTo(bucketStats.criticalCount));
+            assertThat(response.buckets().size(), equalTo((int) Math.min(100, bucketStats.criticalCount)));
+            assertThat(response.buckets().stream().anyMatch(b -> b.getAnomalyScore() < 75.0), is(false));
+        }
+        {
+            GetBucketsRequest request = new GetBucketsRequest(JOB_ID);
+            request.setExcludeInterim(true);
+
+            GetBucketsResponse response = execute(request, machineLearningClient::getBuckets, machineLearningClient::getBucketsAsync);
+
+            assertThat(response.count(), equalTo(240L));
+        }
+        {
+            GetBucketsRequest request = new GetBucketsRequest(JOB_ID);
+            request.setStart("1533081600000");
+            request.setEnd("1533092400000");
+
+            GetBucketsResponse response = execute(request, machineLearningClient::getBuckets, machineLearningClient::getBucketsAsync);
+
+            assertThat(response.count(), equalTo(3L));
+            assertThat(response.buckets().get(0).getTimestamp().getTime(), equalTo(1533081600000L));
```

Comment:
Nit: extract 1533081600000L to a constant FIRST_BUCKET_TIME as the literal is used several times